File: angular/src/app/interfaces/ingredient-meta-data.ts
// deprecated
export interface IngredientMetaData {
  verbose_names: VerboseNames;
  // blank_fields: BlankFields;
  // max_length: MaxLength;
  choices: Choices;
}

export interface VerboseNames {
  [key: string]: string;
  name: string;
  brand: string;
  unit: string;
  kcal: string;
  carbs: string;
  protein: string;
  fat: string;
}

export interface UnitChoices {
  [key: string]: string;
}

// export interface BlankFields {
//   name: boolean;
//   brand: boolean;
//   unit: boolean;
//   kcal: boolean;
//   carbs: boolean;
//   protein: boolean;
//   fat: boolean;
// }

// export interface MaxLength {
//   name: number;
//   brand: number;
//   unit: number;
// }

export interface Choices {
  unit: Choice[];
}

export interface Choice {
  value: string;
  label: string;
}

File: quarkus/src/main/java/org/acme/IngredientMetaData/IngredientMetaData.java
package org.acme.IngredientMetaData;

import java.util.Map;
import java.util.HashMap;

public class IngredientMetaData {

    public static Map<String, String> getVerboseNames() {
        Map<String, String> verboseNames = new HashMap<>();
        verboseNames.put("name", "Name");
        verboseNames.put("brand", "Marke");
        verboseNames.put("unit", "Einheit");        
        verboseNames.put("unitVerbose", "Einheit");
        verboseNames.put("kcal", "Kalorien");
        verboseNames.put("carbs", "Kohlenhydrate");
        verboseNames.put("protein", "Proteine");
        verboseNames.put("fat", "Fett");
        return verboseNames;
    }

    public static Map<String, String> getUnitChoices() {
        Map<String, String> unitChoices = new HashMap<>();
        unitChoices.put("G", "g");
        unitChoices.put("ML", "ml");
        unitChoices.put("PIECE", "Stk.");
        return unitChoices;
    }

    public IngredientMetaData() {}
}